check_test.go

```diff
@@ -0,0 +1,134 @@
+package subtest_test
+
+import (
+	"testing"
+
+	"github.com/searis/subtest"
+)
+
+func TestDeepEqual(t *testing.T) {
+	// Demonstrates GWT-style sub-tests.
+
+	t.Run("when checking bool(true)", func(t *testing.T) {
+		v := true
+		vf := subtest.Value(v)
+
+		t.Run("then it should match true", vf.DeepEqual(true))
+		t.Run("then it should match true", subtest.DeepEqual(true).Test(v)) // equivalent
+
+		expectErr := subtest.Failure{Prefix: "values are not deep equal", Got: "bool\n\ttrue", Expect: "bool\n\tfalse"}
+		t.Run("then a test against false should fail",
+			subtest.Value(subtest.DeepEqual(false)(v)).ErrorIs(expectErr),
+		)
+
+		w := new(bool)
+		*w = true
+
+		expectErr = subtest.Failure{Prefix: "values are not deep equal", Got: "bool\n\ttrue", Expect: "*bool\n\ttrue"}
+		t.Run("then a test against &true should fail",
+			subtest.Value(subtest.DeepEqual(w)(v)).ErrorIs(expectErr),
+		)
+	})
+
+	t.Run("given a nested struct type T", func(t *testing.T) {
+		type T struct {
+			A string
+			B map[string]string
+		}
+
+		t.Run("when checking a non-nil *T value", func(t *testing.T) {
+			v := &T{A: "a", B: map[string]string{"C": "D"}}
+
+			t.Run("then it should match an equivalent *T value",
+				subtest.Value(v).DeepEqual(&T{A: "a", B: map[string]string{"C": "D"}}),
+			)
+			t.Run("then it should not match a different *T value",
+				subtest.Value(subtest.DeepEqual(&T{A: "a", B: map[string]string{"C": "E"}})(v)).Error(),
+			)
+			t.Run("then it should not match an equivalent T value",
+				subtest.Value(subtest.DeepEqual(T{A: "a", B: map[string]string{"C": "D"}})(v)).Error(),
+			)
+
+		})
+	})
+
+}
+
+func TestNotDeepEqual(t *testing.T) {
+	// Demonstrates GWT-style sub-tests.
+
+	t.Run("when checking bool(true)", func(t *testing.T) {
+		v := true
+		vf := subtest.Value(v)
+
+		t.Run("then it should accept false", vf.NotDeepEqual(false))
+		t.Run("then it should accept false", subtest.NotDeepEqual(false).Test(v)) // equivalent
+
+		w := new(bool)
+		*w = true
+		t.Run("then a test against &true should pass", vf.NotDeepEqual(w))
+
+		expectErr := subtest.Failure{Prefix: "values are deep equal", Got: "bool\n\ttrue", Reject: "bool\n\ttrue"}
+		t.Run("then a test against false should fail",
+			subtest.ErrorIs(expectErr).Test(subtest.NotDeepEqual(true)(v)),
```

Comment:
The ordering of calls here makes this a bit weird to look at. I'd expect it to go like this:
`subtest.Test(subtest.NotDeepEqual(true)(v)).ErrorIs(expectErr)`